packages/gatsby/src/utils/handle-flags.ts

```diff
@@ -122,8 +122,14 @@ const handleFlags = (
       flag.includedFlags.forEach(includedName => {
         const incExp = flags.find(e => e.name == includedName)
         if (incExp) {
-          enabledConfigFlags.push(incExp)
-          addIncluded(incExp)
+          const flagIsDisabledByUser =
+            typeof configFlags[includedName] !== `undefined` &&
+            !configFlags[includedName]
+
+          if (!flagIsDisabledByUser) {
+            enabledConfigFlags.push(incExp)
+            addIncluded(incExp)
+          }
```

Comment:
additional checks to see wether user specifaclly disabled certain flag in config before adding it